cime_config/stream_cdeps.py

```diff
@@ -187,7 +188,15 @@ def create_stream_xml(
                     {"name": "NEON.NEON_PRECIP.$NEONSITE"},
                     err_msg="No stream_entry {} found".format(stream_name),
                 )
+            elif stream_name.startswith("PLUMBER2"):
+                self.stream_nodes = super(StreamCDEPS, self).get_child(
+                    "stream_entry",
+                    {"name": "PLUMBER2.$PLUMBER2SITE"},
+                    err_msg="No stream_entry {} found".format(stream_name),
+                )
             elif stream_name.startswith("CLM_USRDAT."):
+                if 'PLUMBER2' in stream_name:
+                    continue
```

Comment:
I'm wondering if this check is needed/correct... if so, it could warrant a comment. What I'm seeing is that the above block (lines 191-196) handles the PLUMBER2 cases, but only for stream names that *start* with PLUMBER2... and then this check on lines 198-199 skips handling in cases where the stream name starts with CLM_USRDAT and contains PLUMBER2 somewhere else in the stream name. Are there indeed cases like this? If so, what happens in this situation? i.e., how is `self.stream_nodes` set? Or does it not need to be set? (I'm also wondering why this handling for PLUMBER would differ from the handling for NEON.)

These are all naive questions that come from not really understanding this logic. Which is to say: I'm very willing to believe that the logic here is correct, but it feels unintuitive to me, so if it is correct, I think a comment explaining the rationale could help.

Reply:
@TeaganKing since it needs it based on your testing keep it in and add a comment about it.

But, man looking it over I am just not seeing why it's needed here. The "elif" should only match PLUMBER2 and not allow it to go into this one. If anyone can figure this out on why it's needed, I'd love to hear...

Reply:
We were ending up with a stream `CLM_USRDAT.PLUMBER2` (in addition to `PLUMBER2.PLUMBER2SITE`.

Reply:
I've been staring at this for a while trying to figure out what might be going on. Major caveat that I don't have a very good understanding of this logic!!!

I'm wondering if the underlying issue is that we're getting an extra / duplicate addition of the plumber streams to the stream list. This led me to wonder if the added lines 202-203 in buildnml are actually correct / needed. @TeaganKing is it possible that those lines aren't needed? Would it be easy to test with both those lines removed and the above lines 198-199 that we're discussing also removed, and see if things work right in that situation? If this testing would be time-consuming, I guess I'm okay leaving this in with a comment. Though I guess I'd also be interested in seeing a print-out of all the stream names in the `stream_names` list just to get a better understanding of what's going on here.

Reply:
@TeaganKing I wrote my comment before seeing your latest. I'm now doubting my guess: It seems like lines 202-203 are probably what add the desired `PLUMBER2.PLUMBER2SITE`... does that seem right to you? If so, maybe the best thing is just to add a comment summarizing this... something along the lines of your last comment explaining that we get this extra `CLM_USRDAT.PLUMBER2` in addition to the desired `PLUMBER2.PLUMBER2SITE` is probably sufficient.

Reply:
Ack, lots of crossed messages this morning where I keep writing mine before seeing your replies - sorry! I guess if it's quick to look into it that would be interesting... but if it would be time-consuming or that doesn't look like the right answer, then let's just stick with your comment.

Reply:
Maybe give me a half hour or so to run a case and test that out.

Reply:
Ok this is failing when I comment out both of those lines... I think it's because `CLM_USRDAT.PLUMBER2` doesn't work (which becomes the stream name when those lines are commented out) but `PLUMBER2.PLUMBER2SITE` does. I think the best method forward is to leave that if statement in place with the comment that I added describing the reason for its existence. Does that work for you?

Reply:
Yes, sounds good. Thanks a lot for testing this and for your excellent communication on this!!!